src/builtins.rs

```diff
@@ -64,7 +72,44 @@ pub fn export(args: Vec<&str>) -> Result<String, Box<dyn Error>> {
 
 pub fn unset(args: Vec<&str>) -> Result<String, Box<dyn Error>> {
     for arg in args {
-        unsafe { std::env::remove_var(arg) };
+        std::env::remove_var(arg);
     }
     Ok("".to_string())
 }
+
+pub fn alias(
+    args: Vec<&str>,
+    aliases: &mut HashMap<String, String>,
+) -> Result<String, Box<dyn Error>> {
+    if args.len() != 1 {
+        return Err("alias: not enough arguments or too many arguments".into());
+    }
+
+    let alias_input = args[0];
+    let parts: Vec<&str> = alias_input.splitn(2, '=').collect();
+
+    if parts.len() != 2 {
+        return Err("alias: invalid format, expected name=value".into());
+    }
+
+    let alias_name = parts[0].to_string();
+    let alias_value = parts[1].to_string();
+
+    // Tokenize the alias value
+    let tokenized = tokenize::tokenize(&mut alias_value.clone())?;
+    let tokenized_value = tokenized.to_string();
+
+    aliases.insert(alias_name, tokenized_value);
+    Ok("".to_string())
+}
+
+pub fn expand_aliases(tokens: &mut Vec<String>, aliases: &HashMap<String, String>) {
+    if let Some(first_token) = tokens.first() {
+        if let Some(alias_value) = aliases.get(first_token) {
+            let mut new_tokens: Vec<String> =
+                alias_value.split_whitespace().map(String::from).collect();
+            new_tokens.extend(tokens.drain(1..));
+            *tokens = new_tokens;
+        }
+    }
+}
```

Comment:
I feel like this should just return the new set of tokens. Since you're already creating a new `vec` of tokens `new_tokens`.